src/instance_launcher.ts

```diff
@@ -115,17 +114,37 @@ export default class InstanceLauncher {
         return true;
     }
 
-    async getInstancesForScaleDown(
+    getJVBsForScaleDown(
         ctx: Context,
-        currentInventory: Array<JibriState>,
         group: InstanceGroup,
-    ): Promise<Array<InstanceDetails>> {
-        const desiredScaleDownQuantity =
-            currentInventory.length - Math.max(group.scalingOptions.minDesired, group.scalingOptions.desiredCount);
+        unprotectedInstances: Array<InstanceState>,
+        desiredScaleDownQuantity: number,
+    ): Array<InstanceDetails> {
+        // first sort by participant count
+        unprotectedInstances.sort((a, b) => {
+            return a.status.jvbStatus.participants - b.status.jvbStatus.participants;
+        });
+        if (unprotectedInstances.length < desiredScaleDownQuantity) {
+            const groupName = group.name;
+            const actualScaleDownQuantity = unprotectedInstances.length;
+            ctx.logger.error(
+                '[Launcher] Nr of JVB instances in group for scale down is less than desired scale down quantity',
+                { groupName, actualScaleDownQuantity, desiredScaleDownQuantity },
+            );
+            desiredScaleDownQuantity = actualScaleDownQuantity;
```

Comment:
I would just use the actualScaleDownQuantity to hold the count of instances to scale down. This is Math.min(unprotectedInstances.length, desiredScaleDownQuantity)